### PR TITLE
Fix #2479

### LINKED ISSUE #2479: Formatting issues with as-patterns inside implicit params
The formatter generates the following:

```
a : {A : Type} → A × Nat → Nat
-  | {A'@_} p@(x, s@zero) := snd p + 1
+  | A'@{_} p@(x, s@zero) := snd p + 1
  | p@(x, s@_) := snd p + 1;
```

- Blocking https://github.com/anoma/juvix/pull/2404

### PR DESCRIPTION
**Pull request by jonaprieto**
Fix formatting for as-patterns inside implicit patterns.

### CODE REVIEW

File: tests/positive/Format.juvix [hunk 1]
```diff
@@ -375,3 +375,15 @@ l4 : List (List Int) :=
   ];
 
 -- Comment at the end of a module
```
Comment (paulcadman) [inline tests/positive/Format.juvix:377]: Move this line (`-- Comment at the end of a module`) to the end of the module.

File: src/Juvix/Compiler/Concrete/Print/Base.hs [hunk 2]
```diff
@@ -1014,11 +1014,18 @@ instance PrettyPrint PatternArg where
   ppCode PatternArg {..} = do
     let name' = ppCode <$> _patternArgName
         pat' = ppCode _patternArgPattern
-    (name' <&> (<> ppCode Kw.kwAt))
-      ?<> delimIf _patternArgIsImplicit delimCond pat'
-    where
-      delimCond :: Bool
-      delimCond = isJust _patternArgName && not (isAtomic _patternArgPattern)
+
+    let delimCond :: Bool
```
Comment (paulcadman) [inline src/Juvix/Compiler/Concrete/Print/Base.hs:1018]: These lets can be combined into a single let block.

Comment (jonaprieto) [reply src/Juvix/Compiler/Concrete/Print/Base.hs:1018]: Done

File: tests/positive/Format.juvix (hunk 1 above)
Comment (jonaprieto) [reply tests/positive/Format.juvix]: Done

**Review by paulcadman — APPROVED** [review]

**Pull request merged**